system/datatypes/peer_reviewed_article.php

```diff
@@ -374,6 +387,176 @@ protected function defaultView(HyphaRequest $request) {
+				},
+				'level' => self::CHECK_LEVEL_REQUIRED,
+			],
+			'has-method' => [
+				'statuses' => [self::STATUS_DRAFT, self::STATUS_REVIEW, self::STATUS_APPROVED],
+				'checker' => function() {
+					return [$this->xml->find(self::FIELD_NAME_METHOD)->getText() !== "", []];
+				},
+				'level' => self::CHECK_LEVEL_REQUIRED,
+			],
+			'has-sources' => [
+				'statuses' => [self::STATUS_DRAFT, self::STATUS_REVIEW, self::STATUS_APPROVED],
+				'checker' => function() {
+					return [$this->xml->find(self::FIELD_NAME_SOURCES)->getText() !== "", []];
+				},
+				'level' => self::CHECK_LEVEL_RECOMMENDED,
+			],
+			'has-excerpt' => [
+				'statuses' => [self::STATUS_DRAFT, self::STATUS_REVIEW, self::STATUS_APPROVED],
+				'checker' => function() {
+					return [$this->getExcerpt()->getText() !== "", []];
+				},
+				'level' => self::CHECK_LEVEL_REQUIRED,
+			],
+			'has-featured-image' => [
+				'statuses' => [self::STATUS_DRAFT, self::STATUS_REVIEW, self::STATUS_APPROVED],
+				'checker' => function() {
+					return [$this->xml->find(self::FIELD_NAME_FEATURED_IMAGE)->getText() !== "", []];
+				},
+				'level' => self::CHECK_LEVEL_RECOMMENDED,
+			],
+			'has-tags' => [
+				'statuses' => [self::STATUS_DRAFT, self::STATUS_REVIEW, self::STATUS_APPROVED],
+				'checker' => function() {
+					return [count(HyphaTags::tagsForPageListNode($this->pageListNode)) > 0, []];
+				},
+				'level' => self::CHECK_LEVEL_REQUIRED,
+			],
+			'has-approvals' => [
+				'statuses' => [self::STATUS_REVIEW, self::STATUS_APPROVED],
+				'checker' => function() {
+					$approveCount = $this->xml->find(self::FIELD_NAME_APPROVE_CONTAINER)->children()->count();
+					return [$approveCount >= self::APPROVES_NEEDED, ['have' => $approveCount, 'need' => self::APPROVES_NEEDED]];
+				},
+				'level' => self::CHECK_LEVEL_REQUIRED,
+			],
+			'has-no-blocks' => [
+				'statuses' => [self::STATUS_REVIEW, self::STATUS_APPROVED],
+				'checker' => function() {
+					$xpath = '//' . self::FIELD_NAME_DISCUSSION_CONTAINER . '//' . self::FIELD_NAME_DISCUSSION . '[@' . self::FIELD_NAME_DISCUSSION_BLOCKING . '="1" and @' . self::FIELD_NAME_DISCUSSION_BLOCK_RESOLVED . '!="1"]';
+					$blockingCount = $this->xml->findXPath($xpath)->count();
+					return [$blockingCount === 0, ['left' => $blockingCount]];
+				},
+				'level' => self::CHECK_LEVEL_REQUIRED,
+			],
+		];
+	}
+
+	/**
+	 * Runs any checks, as appropriate for the current status, and
+	 * returns both the overall result, and a list of each check for
+	 * the current status with its result.
+	 *
+	 * @return array{
+	 *   (self::CHECK_STATUS_*),
+	 *   array<string, array{
+	 *     passed: bool,
+	 *     status: self::CHECK_STATUS_*,
+	 *     label: string,
+	 *   }>}
+	 */
+	protected function runChecks() {
+		$result = [];
+		$overallStatus = self::CHECK_STATUS_PASSED;
+
+		foreach ($this->getChecks() as $name => $check) {
+			if (!in_array($this->getStatus(), $check['statuses']))
```

Comment:
The status gotten from `getStatus()` does not change while being in this method, so it would be better to assign in to a variable outside the foreach loop.

Reply:
Good call, fixed.